components/NavigationBar.tsx

```diff
@@ -0,0 +1,171 @@
+'use client';
+
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import Link from 'next/link';
+import Image from 'next/image';
+import { usePathname } from 'next/navigation';
+import { useTheme } from 'next-themes';
+
+import ThemeSwitcher from './ThemeSwitcher';
+import { close, open, openWhite, closeWhite } from '../public/assets/index';
+import { logo } from '@/public/assets-png';
+
+import DownloadResume from './Download';
+
+const routes = [
+  { path: '/', label: 'Home' },
+  { path: '/case-studies', label: 'Case Studies' },
+  { path: '/contact', label: 'Contacts' },
+];
+
+const Navbar: React.FC = () => {
+  const [showMobileNavbar, setShowMobileNavbar] = useState(false);
+  const { theme } = useTheme();
+  const openImgSrc = theme === 'dark' ? openWhite : open;
+  const closeImgSrc = theme === 'dark' ? closeWhite : close;
+  const pathname = usePathname();
+
+  return (
+    <div className="fixed z-50 flex h-16 w-full justify-between bg-white-800 py-[2.2rem] dark:bg-black-200">
+      <Link
+        href={{
+          pathname: '/',
+        }}
+        className="ml-[85px] flex items-center"
+      >
+        <div className="flex items-center">
+          <Image
+            src={logo}
+            width={30}
+            height={30}
+            alt="logo"
+            className="md:h-[45px] md:w-[45px]"
+          />
+        </div>
+      </Link>
+      <DesktopNavbar
+        setShowMobileNavbar={setShowMobileNavbar}
+        showMobileNavbar={showMobileNavbar}
+        openImgSrc={openImgSrc}
+        pathname={pathname}
+      />
+
+      {showMobileNavbar && (
+        <MobileNavigation
+          setShowMobileNavbar={setShowMobileNavbar}
+          showMobileNavbar={showMobileNavbar}
+          closeImgSrc={closeImgSrc}
+          pathname={pathname}
+        />
+      )}
+    </div>
+  );
+};
+
+export default Navbar;
+
+type MobileNavigationProps = {
+  setShowMobileNavbar: Dispatch<SetStateAction<boolean>>;
+  showMobileNavbar: boolean;
+  closeImgSrc: string;
+  pathname: string;
+};
+
+type DeskTopNavbarProps = Omit<MobileNavigationProps, 'closeImgSrc'> & {
+  openImgSrc: string;
+};
+
+type NavigationRoutesProps = {
+  pathname: string;
+  showMobileNavbar: boolean;
+};
+
+const NavigationRoutes: React.FC<NavigationRoutesProps> = ({
+  pathname,
+  showMobileNavbar,
+}) => {
+  return (
+    <>
+      {routes.map((route, index) => (
+        <Link href={route.path} key={index}>
+          <p
+            className={`mx-3.5 ${
+              showMobileNavbar ? 'mt-6' : ''
+            }  flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
+              pathname === route.path &&
+              'font-semibold text-[#0252CD] dark:text-[#428DFF]'
+            }`}
+          >
+            {route.label}
+          </p>
+        </Link>
+      ))}
```

Comment:
```suggestion
      {routes.map((route) => (
        <Link href={route.path} key={route.label}>
          <p
            className={`mx-3.5 ${
              showMobileNavbar ? 'mt-6' : ''
            }  flex h-full items-center text-[0.875rem] font-normal text-white-500 dark:text-white-800 ${
              pathname === route.path &&
              'font-semibold text-[#0252CD] dark:text-[#428DFF]'
            }`}
          >
            {route.label}
          </p>
        </Link>
      ))}
```

Key's should reflect something unique about the element when possible.

Reply:
Resolved